contracts/DomainRegistry.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+/// @title Domain Registry Contract
+/// @notice This contract allows users to register and manage top-level domain names
+/// @author developed by Ivan Myasoyedov
+contract DomainRegistry {
+    /// @notice Address of the contract owner
+    address public owner;
+
+    /// @notice Fee required to register a domain
+    uint256 public registrationFee;
+
+    /// @dev Struct to store information about a domain
+    struct Domain {
+        string name;
+        address controller;
+        uint256 registeredAt;
+    }
+
+    /// @dev Array of all domains registered in the contract
+    Domain[] public domains;
+
+    /// @dev Mapping from domain name to controller address
+    mapping(string => address) public domainToController;
+
+    /// @dev Mapping from controller address to their domains
+    mapping(address => Domain[]) public controllerToDomains;
+
+    /// @notice Emitted when a new domain is registered
+    event DomainRegistered(
+        string domain,
+        address indexed controller,
+        uint256 timestamp
+    );
+
+    /// @dev Ensures that only the owner can call the function
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Only the owner can call this function");
+        _;
+    }
+
+    /// @notice Creates a new DomainRegistry contract
```

Comment:
line 43-44 and other same places
NatSpec comments for multiple lines should be written with /** ... */
// - 1 point